File: src/main/java/com/gtladd/gtladditions/Common/Data/GTLAddCreativeModeTabs.java
package com.gtladd.gtladditions.Common.Data;

import org.gtlcore.gtlcore.common.data.machines.AdvancedMultiBlockMachine;

import com.gregtechceu.gtceu.common.data.GTCreativeModeTabs;

import net.minecraft.world.item.CreativeModeTab;

import com.gtladd.gtladditions.Common.Items.GTLAddItems;
import com.gtladd.gtladditions.GTLAdditions;
import com.tterrag.registrate.util.entry.RegistryEntry;

import static com.gtladd.gtladditions.api.Registry.GTLAddRegistration.REGISTRATE;

public class GTLAddCreativeModeTabs {

    public static RegistryEntry<CreativeModeTab> GTLADD_ITEMS = REGISTRATE.defaultCreativeTab("item",
            builder -> builder.displayItems(new GTCreativeModeTabs.RegistrateDisplayItemsGenerator("item", REGISTRATE))
                    .title(REGISTRATE.addLang("itemGroup", GTLAdditions.id("item"), "GTLAdditions"))
                    .icon(GTLAddItems.STARMETAL_BOULE::asStack)
                    .build())
            .register();
    public static RegistryEntry<CreativeModeTab> GTLADD_MACHINE = REGISTRATE.defaultCreativeTab("machine",
            builder -> builder.displayItems(new GTCreativeModeTabs.RegistrateDisplayItemsGenerator("machine", REGISTRATE))
                    .title(REGISTRATE.addLang("itemGroup", GTLAdditions.id("machine"), "GTLAdditions"))
                    .icon(AdvancedMultiBlockMachine.EYE_OF_HARMONY::asStack)
                    .build())
            .register();

    public static void init() {}
}

File: src/main/java/com/gtladd/gtladditions/common/data/GTLAddCreativeModeTabs.kt
package com.gtladd.gtladditions.common.data

import com.gregtechceu.gtceu.common.data.GTCreativeModeTabs.RegistrateDisplayItemsGenerator
import com.gtladd.gtladditions.GTLAdditions
import com.gtladd.gtladditions.api.registry.GTLAddRegistration.REGISTRATE
import com.gtladd.gtladditions.common.blocks.GTLAddBlocks
import com.gtladd.gtladditions.common.items.GTLAddItems
import com.tterrag.registrate.util.entry.RegistryEntry
import net.minecraft.world.item.CreativeModeTab
import org.gtlcore.gtlcore.common.data.machines.AdvancedMultiBlockMachine

object GTLAddCreativeModeTabs {
    @JvmField
    val GTLADD_ITEMS: RegistryEntry<CreativeModeTab?> = REGISTRATE.defaultCreativeTab("item")
    { builder : CreativeModeTab.Builder? ->
        builder !!.displayItems(RegistrateDisplayItemsGenerator("item", REGISTRATE))
            .title(
                REGISTRATE.addLang("itemGroup", GTLAdditions.id("item"), "GTLAdditions")
            )
            .icon { GTLAddItems.STARMETAL_BOULE.asStack() }
            .build()
    }
        .register()
    @JvmField
    val GTLADD_MACHINE: RegistryEntry<CreativeModeTab?> = REGISTRATE.defaultCreativeTab("machine")
    { builder : CreativeModeTab.Builder? ->
        builder !!.displayItems(RegistrateDisplayItemsGenerator("machine", REGISTRATE))
            .title(
                REGISTRATE.addLang("itemGroup", GTLAdditions.id("machine"), "GTLAdditions")
            )
            .icon { AdvancedMultiBlockMachine.EYE_OF_HARMONY.asStack() }
            .build()
    }
        .register()

    @JvmField
    val GTLADD_BLOCKS: RegistryEntry<CreativeModeTab?> = REGISTRATE.defaultCreativeTab("block")
    { builder : CreativeModeTab.Builder? ->
        builder !!.displayItems(RegistrateDisplayItemsGenerator("block", REGISTRATE))
            .title(
                REGISTRATE.addLang("itemGroup", GTLAdditions.id("block"), "GTLAdditions")
            )
            .icon { GTLAddBlocks.QUANTUM_GLASS.asStack() }
            .build()
    }
        .register()

    @JvmStatic
    fun init() {}
}
